sink/sink

```diff
@@ -412,17 +416,76 @@ def mkdir_and_chdir(base, identifier):
     os.fchmod(dirfd, 0o755)
     os.close(dirfd)
 
+def prune_logs(base_path=".", dry_run=False):
+    """ Check all subdirectories of the current path and delete the ones that
+        are either empty or contain only files older than the age threshold.
+
+        For dry runs, print the name of the directory instead.
+    """
+    reference_time = time.time()
+    age_threshold = 30
+    entries = os.listdir(base_path)
```

Comment:
This will include files, such as the `.prune` stamp file.  I think we should skip non-directories.